### PR TITLE
[HWORKS-888] Improve documentation for gRPC support

### PR DESCRIPTION
**Pull request by javierdlrm**

### CODE REVIEW

File: docs/concepts/mlops/kserve.md [hunk 1]
```diff
@@ -1,19 +1,30 @@
...
- - **Inference Batcher**
-    - Inference requests can be batched in different ways to adjust the trade-off between throughput and latencies of the model predictions
- - **Versioned Deployments**
-    - Model deployments are versioned, enabling A/B testing and more.
- - **Istio Model Endpoint**
-    - You can publish model via a REST Endpoint using Istio and access it over HTTP using a Hopsworks API key (with serving scope). Secure and authorized access is guaranteed by Hopsworks.
-
-Models deployed on KServe in Hopsworks can be easily integrated with the Hopsworks feature store using a Transformer Python script, that builds the predictor's input feature vector using the application input and pre-computed features from the feature store.
+In Hopsworks, you can easily deploy models from the model registry in KServe or in Docker containers (for Hopsworks Community). KServe is an open-source framework for model serving on Kubernetes. You can deploy models in either programs, using the HSML library, or in the UI. A KServe model deployment can include the following components:
```
Comment (jimdowling) [inline docs/concepts/mlops/kserve.md:1]: is the defacto open-source framework

File: docs/user_guides/mlops/serving/api-protocol.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+<p align="center">
+  <figure>
+    <img src="../../../../assets/images/guides/mlops/serving/deployment_adv_form_kserve.png" alt="KServe enabled in advanced deployment form">
+    <figcaption>Enable KServe in the advanced deployment form</figcaption>
+  </figure>
+</p>
+
+Then, you can select the API protocol to be enabled in your model deployment.
+
+!!! note "Only one API protocol can be enabled simultaneously"
```
Comment (gibchikafa) [inline docs/user_guides/mlops/serving/api-protocol.md:49]: Do we need to say simultaneously?

File: docs/concepts/mlops/kserve.md [hunk 3]
```diff
@@ -1,19 +1,30 @@
...
+
+:   A predictor is a ML model in a Python object that takes a feature vector as input and returns a prediction as output.
+
+**`Inference Logger`**
+
+:   Hopsworks logs inputs and outputs of transformers and predictors to a ^^Kafka topic^^ that is part of the same project as the model.
+
+**`Inference Batcher`**
+
+:   Inference requests can be batched in different ways to adjust the trade-off between throughput and latencies of the model predictions.
```
Comment (jimdowling) [inline docs/concepts/mlops/kserve.md:18]: Inference requests can be batched to improve throughput (at the cost of slightly higher latency).

File: docs/concepts/mlops/kserve.md [hunk 4]
```diff
@@ -1,19 +1,30 @@
...
+
+**`Istio Model Endpoint`**
+
+:   You can publish a model via a ^^REST or gRPC Endpoint^^ using Istio and access it over HTTP using a Hopsworks API key (with serving scope). Secure and authorized access is guaranteed by Hopsworks.
```
Comment (jimdowling) [inline docs/concepts/mlops/kserve.md:22]: over HTTP or gRPC using a Hopsworks API key.  API keys have scopes to ensure the principle of least privilege access control to resources managed by Hopsworks.

**Comment by gibchikafa** [comment]
Don't we need to explain about the InferInput, especially the data types and shape?

File: docs/concepts/mlops/kserve.md [hunk 5]
```diff
@@ -1,19 +1,30 @@
...
+
+
+Models deployed on KServe in Hopsworks can be easily integrated with the Hopsworks Feature Store using a Transformer Python script, that builds the predictor's input feature vector using the application input and pre-computed features from the Feature Store.
```
Comment (jimdowling) [inline docs/concepts/mlops/kserve.md:25]: using either a Transformer or Predictor Python script

File: docs/user_guides/mlops/serving/api-protocol.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
+# How to Select the API protocol for a Deployment
+
+## Introduction
+
+Hopsworks supports both REST and gRPC as API protocols for sending inference requests to model deployments. While REST API protocol is supported in all types of model deployments, support for gRPC is only available for models served with [KServe](predictor.md#serving-tool).
+
+!!! warning
+    At the moment, gRPC API protocol is only supported for **Python model deployments** (e.g., scikit-learn, xgboost, ...). Support for Tensorflow model deployments is coming soon.
```
Comment (jimdowling) [inline docs/user_guides/mlops/serving/api-protocol.md:8]: At the moment, the gRPC API protocol

File: docs/user_guides/mlops/serving/api-protocol.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+## GUI
+
+### Step 1: Create new deployment
```
Comment (jimdowling) [inline docs/user_guides/mlops/serving/api-protocol.md:12]: Create a new deployment

File: docs/user_guides/mlops/serving/api-protocol.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+If you have at least one model already trained and saved in the Model Registry, navigate to the deployments page by clicking on the `Deployments` tab on the navigation menu on the left.
+
+<p align="center">
+  <figure>
+    <img src="../../../../assets/images/guides/mlops/serving/deployments_tab_sidebar.png" alt="Deployments navigation tab">
+    <figcaption>Deployments navigation tab</figcaption>
+  </figure>
+</p>
+
+Once in the deployments page, click on `New deployment` if there are not existing deployments or on `Create new deployment` at the top-right corner to open the deployment creation form.
```
Comment (jimdowling) [inline docs/user_guides/mlops/serving/api-protocol.md:23]: Once in the deployments page, you can create a new deployment by either clicking on `New deployment` (if there are no existing deployments) or on `Create new deployment` it the top-right corner. Both options will open the deployment creation form.

File: docs/user_guides/mlops/serving/api-protocol.md [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+
+### Step 2: Go to advanced options
+
+A simplified creation form will appear including the most common deployment fields among all the configuration possible. Resource allocation is part of the advanced options of a deployment. To navigate to the advanced creation form, click on `Advanced options`.
```
Comment (jimdowling) [inline docs/user_guides/mlops/serving/api-protocol.md:27]: from all available configurations.

File: docs/user_guides/mlops/serving/api-protocol.md (hunk 2 above)
Comment (jimdowling) [inline docs/user_guides/mlops/serving/api-protocol.md:49]: Only one API protocol can be enabled for a model (they cannot support both gRPC and REST)

File: docs/user_guides/mlops/serving/api-protocol.md [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+  <figure>
+    <img src="../../../../assets/images/guides/mlops/serving/deployment_adv_form_kserve.png" alt="KServe enabled in advanced deployment form">
+    <figcaption>Enable KServe in the advanced deployment form</figcaption>
+  </figure>
+</p>
+
+Then, you can select the API protocol to be enabled in your model deployment.
+
+!!! note "Only one API protocol can be enabled simultaneously"
+    Currently, KServe model deployments are limited to one API protocol at a time. Therefore, only one of REST or gRPC API protocols can be enabled at the same time on the same model deployment.
```
Comment (jimdowling) [inline docs/user_guides/mlops/serving/api-protocol.md:50]: After a model deployment has been created, you cannot change its API protocol - you will need to create a new deployment.

File: docs/user_guides/mlops/serving/deployment.md [hunk 11]
```diff
@@ -192,6 +193,10 @@ Inference batcher are deployment component that apply batching to the incoming i
 
 Resources include the number of replicas for the deployment as well as the resources (i.e., memory, CPU, GPU) to be allocated per replica. To learn about the different combinations available, see the [Resources Guide](resources.md).
 
+## API protocol
+
+Hopsworks supports both REST and gRPC as the API protocols to send inference requests to model deployments. To learn about REST and gRPC API protocols for model deployments, see the [API Protocol Guide](api-protocol.md).
```
Comment (jimdowling) [inline docs/user_guides/mlops/serving/deployment.md:198]: In general, you use gRPC when you need lower latency inference requests. To learn more about the REST and gRPC API protocols

File: docs/user_guides/mlops/serving/predictor.md [hunk 12]
```diff
@@ -246,6 +248,10 @@ Inference batcher are deployment component that apply batching to the incoming i
 
 Resources include the number of replicas for the deployment as well as the resources (i.e., memory, CPU, GPU) to be allocated per replica. To learn about the different combinations available, see the [Resources Guide](resources.md).
 
+## API protocol
```
Comment (jimdowling) [inline docs/user_guides/mlops/serving/predictor.md:251]: This is duplicate - see line 196

**Review by jimdowling — APPROVED** [review]
Please fix the changes before merging

**Comment by javierdlrm** [comment]
> Don't we need to explain about the InferInput, especially the data types and shape?

Yes, I'm planning to write a new whole page focused on model inference with hsml.

File: docs/user_guides/mlops/serving/api-protocol.md (hunk 2 above)
Comment (javierdlrm) [reply docs/user_guides/mlops/serving/api-protocol.md:49]: To me there's a slight difference between "only one API protocol can be enabled" and "only one API protocol can be enabled simultaneously" :)  the former is not totally correct.

File: docs/user_guides/mlops/serving/api-protocol.md (hunk 10 above)
Comment (javierdlrm) [reply docs/user_guides/mlops/serving/api-protocol.md:50]: You can change the API protocol of an existing deployment.

**Pull request merged**